tokio-console/src/conn.rs

```diff
@@ -128,20 +143,39 @@ impl Connection {
         }
     }
 
-    pub async fn next_update(&mut self) -> Update {
+    pub async fn next_message(&mut self) -> Message {
         loop {
-            match self.state {
-                State::Connected { ref mut stream, .. } => match stream.next().await {
-                    Some(Ok(update)) => return update,
-                    Some(Err(status)) => {
-                        tracing::warn!(%status, "error from stream");
-                        self.state = State::Disconnected(Self::BACKOFF);
+            match &mut self.state {
+                State::Connected {
+                    update_stream,
+                    state_stream,
+                    ..
+                } => {
+                    tokio::select! { biased; // Always biased to update stream.
+                        update = update_stream.next() => match update {
```

Comment:
Is `StreamExt::next` cancel safe? I didn't see the answer immediately in the documentation.

Reply:
Yes. I think it is.
https://docs.rs/tokio/latest/tokio/macro.select.html#cancellation-safety

<img width="1037" alt="image" src="https://github.com/user-attachments/assets/5223c822-4b0f-46a8-88be-1a5a6fbe946c">